Products/GenericSetup/tests/test_tool.py

```diff
@@ -1706,21 +1714,26 @@ def test_getDependenciesForProfile(self):
         site = self._makeSite()
         tool = self._makeOne('setup_tool').__of__(site)
 
-        # Register two extension profiles.  Profile 'foo' has a dependency
-        # 'bar'.
+        # Register three extension profiles.  Profile 'foo' has a dependency
+        # 'bar', and 'baz' contains non-existing dependency-profiles.
         self._makeFile(METADATA_XML, _METADATA_XML)
         _makeTestFile(METADATA_XML, self._PROFILE_PATH2, _PLAIN_METADATA_XML)
+        _makeTestFile(METADATA_XML, self._PROFILE_PATH3, _BROKEN_METADATA_XML)
         profile_registry.registerProfile(
             'foo', 'Foo', '', self._PROFILE_PATH, profile_type=EXTENSION)
         profile_registry.registerProfile(
             'bar', 'Bar', '', self._PROFILE_PATH2, profile_type=EXTENSION)
+        profile_registry.registerProfile(
+            'baz', 'Baz', '', self._PROFILE_PATH3, profile_type=EXTENSION)
 
         self.assertEqual(tool.getDependenciesForProfile('other:foo'),
                          (u'profile-other:bar', ))
         self.assertEqual(tool.getDependenciesForProfile('other:bar'), ())
         self.assertEqual(tool.getDependenciesForProfile('snapshot-some'), ())
         self.assertEqual(tool.getDependenciesForProfile(None), ())
         self.assertRaises(KeyError, tool.getDependenciesForProfile, 'nonesuch')
+        # profile_info does contain dependencies, but one of them doesn't exist:
+        self.assertEqual(tool.dependenciesExist('other:baz'), False)
```

Comment:
This bit really belongs in a separate testcase.  Given that it is here:

- Use `self.assertFalse`.
- Add assertions for the other two profiles.

E.g.:

```python
        # profile_info contains dependencies, and they exist.
        self.assertTrue(tool.dependenciesExist('other:foo'))
        # profile does not contain dependencies
        self.assertTrue(tool.dependenciesExist('other:bar'))
        # profile_info contains dependencies, but one of them doesn't exist:
        self.assertFalse(tool.dependenciesExist('other:baz'))
```

Reply:
Agree. Thanks for the corrections! Gonna be on it tomorrow again.